### ISSUE TITLE
Softmax activation function throws deprecation warning

### ISSUE THREAD

**Issue by vdplasthijs** [qubvel/segmentation_models.pytorch#689]
When defining a `smp` model in `__init__()` as:

```
self.base = smp.Unet(encoder_name='resnet50', pretrained='imagenet', 
                                  in_channels=3, classes=7,
                                  activation='softmax') 
```

This will throw the following warning upon initialisation:

`~/anaconda3/envs/geo/lib/python3.7/site-packages/segmentation_models_pytorch/base/modules.py:116: UserWarning: Implicit dimension choice for softmax has been deprecated. Change the call to include dim=X as an argument.
  return self.activation(x)`

As the softmax is defined by passing 'softmax' as an arg, I'm not sure where/how to include the dim as the warning suggests? Many thanks

See also this closed (but without stating resolution) issue:
_Originally posted by @vdplasthijs in https://github.com/qubvel/segmentation_models.pytorch/issues/169#issuecomment-1334066128_

**Comment by kirillmeisser** [comment]
Hi,

I have fixed this issue by using the "softmax2d" activation function. As you can see in the following snippet it is the same functionality for 2D images but with dimension explicitly set to 1.

https://github.com/qubvel/segmentation_models.pytorch/blob/6db76a1106426ac5b55f39fba68168f3bccae7f8/segmentation_models_pytorch/base/modules.py#L84-L98
